src/Raven.Client/Documents/AI/AiConversation.cs

```diff
@@ -174,6 +174,16 @@ public async Task<AiAnswer<TAnswer>> RunAsync<TAnswer>(CancellationToken token =
                         // error handling here is expected to be done by the invocation based on the error strategy the user choose
                         await invocation.Invoke(ctx, action, token).ConfigureAwait(false);
                     }
+                    else if (OnUnhandledAction is { } e)
+                    {
+                        e(this, action);
```

Comment:
what if this throws an exception? should we send it to the model?

Reply:
No, because when you run into this, we are already in "unknown action".
It's something that we weren't supposed to see? Or maybe the user is handling that?

There is an argument here to making this _async_, though? Thoughts?

Reply:
This would be a single place with async, every where else we are using EventHandler. Can we make it consistent? I do not think it is such an issue to have a sync event here.

Also requires a test to check the behavior.

Reply:
Problem is that you're very likely to _need_ to do async here.
Consider what your typical client code would be.
Your logic says that you show a dialog to the user on such a case - so you may need to send data to the user, which requires async.

We _can_ try to do that using sync code and then gather and send, but nicer to have async here.

Having both sync & async is also an option - but kind of ugly.

I want to see if any tests fail right now because the _are_ getting the new error, by the way (there should be some)

Reply:
Hmm,

then let's wrap the args into a class, this will allow us to extend this without causing any breaking changes.

```
Func<IAiConversationOperations, UnhandledAiAgentActionEventArgs, CancellationToken, Task>
```